core/arch/arm/encode.c

```diff
@@ -1098,7 +1100,8 @@ encode_opnd_ok(decode_info_t *di, byte optype, opnd_size_t size_temp, instr_t *i
     /* Roll back greedy reglist if necessary */
     if (di->reglist_stop > 0 && optype_is_reg(optype) &&
         (!di->reglist_simd || !optype_is_gpr(optype)) &&
-        di->reglist_stop - 1 > di->reglist_start && di->reglist_stop == opnum) {
+        di->reglist_stop - di->reglist_start > di->reglist_min_num &&
```

Comment:
Yes, I noticed that discrepancy. But perhaps we should allow an empty register list in some cases. Here's what the disassembler says about an LDM with an empty register list:
```
	e8900000 	ldm	r0, {}	; <UNPREDICTABLE>
```
So the encoding is perhaps valid, in a sense. It's just very unclear what the instruction would do if we executed it.